packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -410,106 +859,786 @@ class _CupertinoAlertActionSection extends StatelessWidget {
+///
+/// See also:
+///
+///  * [CupertinoAlertDialog], a dialog that informs the user about situations
+///    that require acknowledgement
+class CupertinoDialogAction extends StatelessWidget {
+  /// Creates an action for an iOS-style dialog.
+  const CupertinoDialogAction({
+    this.onPressed,
+    this.isDefaultAction = false,
+    this.isDestructiveAction = false,
+    this.textStyle,
+    @required this.child,
+  }) : assert(child != null);
+
+  /// The callback that is called when the button is tapped or otherwise
+  /// activated.
+  ///
+  /// If this is set to null, the button will be disabled.
+  final VoidCallback onPressed;
+
+  /// Set to true if button is the default choice in the dialog.
+  ///
+  /// Default buttons are bold.
+  final bool isDefaultAction;
+
+  /// Whether this action destroys an object.
+  ///
+  /// For example, an action that deletes an email is destructive.
+  final bool isDestructiveAction;
+
+  /// [TextStyle] to apply to any text that appears in this button.
+  ///
+  /// Dialog actions have a built-in text resizing policy for long text. To
+  /// ensure that this resizing policy always works as expected, [textStyle]
+  /// must be used if a text size is desired other than that specified in
+  /// [_kCupertinoDialogActionStyle].
+  final TextStyle textStyle;
+
+  /// The widget below this widget in the tree.
+  ///
+  /// Typically a [Text] widget.
+  final Widget child;
+
+  /// Whether the button is enabled or disabled. Buttons are disabled by
+  /// default. To enable a button, set its [onPressed] property to a non-null
+  /// value.
+  bool get enabled => onPressed != null;
+
+  double _calculatePadding(BuildContext context) {
+    return 8.0 * MediaQuery.textScaleFactorOf(context);
+  }
+
+  // Dialog action content shrinks to fit, up to a certain point, and if it still
+  // cannot fit at the minimum size, the text content is ellipsized.
+  //
+  // This policy only applies when the device is not in accessibility mode.
+  Widget _buildContentWithRegularSizingPolicy({
+    @required BuildContext context,
+    @required TextStyle textStyle,
+    @required Widget content,
+  }) {
+    final bool isInAccessibilityMode = _isInAccessibilityMode(context);
+    final double dialogWidth = isInAccessibilityMode
+        ? _kAccessibilityCupertinoDialogWidth
+        : _kCupertinoDialogWidth;
+    final double textScaleFactor = MediaQuery.textScaleFactorOf(context);
+    // The fontSizeRatio is the ratio of the current text size (including any
+    // iOS scale factor) vs the minimum text size that we allow in action
+    // buttons. This ratio information is used to automatically scale down action
+    // button text to fit the available space.
+    final double fontSizeRatio = (textScaleFactor * textStyle.fontSize) / _kMinButtonFontSize;
+    final double padding = _calculatePadding(context);
+
+    return new IntrinsicHeight(
+      child: new SizedBox(
+        width: double.infinity,
```

Comment:
i'm pretty sure this SizedBox doesn't do anything useful.

Reply:
Indeed